ipu-plugin/pkg/ipuplugin/vendorservice.go

```diff
@@ -0,0 +1,233 @@
+// Copyright (c) 2024 Intel Corporation.  All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License")
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package ipuplugin
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"os"
+	"strings"
+
+	"github.com/openshift/dpu-operator/api/dpu_api"
+	"github.com/vishvananda/netlink"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+type VendorManagementServer struct {
+	dpu_api.UnimplementedVendorManagementServiceServer
+	ip   string
+	port int
+	mode string
+}
+
+func NewVendorManagementService(ip string, port int, mode string) *VendorManagementServer {
+	return &VendorManagementServer{
+		ip:   ip,
+		port: port,
+		mode: mode,
+	}
+}
+
+const (
+	HOST_VPORT_ID = "03"
+	ACC_VPORT_ID  = "02"
+
+	DEVICE = "0x1452"
+	VENDOR = "0x8086"
+
+	ACC_IP  = "192.168.11.2"
+	HOST_IP = "192.168.11.1"
+	PORT    = 50151
+)
+
+type NetworkHandler interface {
+	AddrAdd(link netlink.Link, addr *netlink.Addr) error
+	AddrList(link netlink.Link, family int) ([]netlink.Addr, error)
+	LinkList() ([]netlink.Link, error)
+}
+
+type NetworkHandlerImpl struct{}
+
+func (h *NetworkHandlerImpl) AddrAdd(link netlink.Link, addr *netlink.Addr) error {
+	return netlink.AddrAdd(link, addr)
+}
+func (h *NetworkHandlerImpl) AddrList(link netlink.Link, family int) ([]netlink.Addr, error) {
+	return netlink.AddrList(link, family)
+}
+func (h *NetworkHandlerImpl) LinkList() ([]netlink.Link, error) {
+	return netlink.LinkList()
+}
+
+type FileSystemHandler interface {
+	GetDevice(iface string) ([]byte, error)
+	GetVendor(iface string) ([]byte, error)
+}
+type FileSystemHandlerImpl struct{}
+
+func (fs *FileSystemHandlerImpl) GetDevice(iface string) ([]byte, error) {
+	return os.ReadFile(fmt.Sprintf("/sys/class/net/%s/device/device", iface))
+}
+func (fs *FileSystemHandlerImpl) GetVendor(iface string) ([]byte, error) {
+	return os.ReadFile(fmt.Sprintf("/sys/class/net/%s/device/vendor", iface))
+}
+
+var fileSystemHandler FileSystemHandler
+var networkHandler NetworkHandler
+
+func initHandlers() {
+	if fileSystemHandler == nil {
+		fileSystemHandler = &FileSystemHandlerImpl{}
+	}
+	if networkHandler == nil {
+		networkHandler = &NetworkHandlerImpl{}
+	}
+}
+
+func isPF(iface string) bool {
+	device, _ := fileSystemHandler.GetDevice(iface)
+	vendor, _ := fileSystemHandler.GetVendor(iface)
```

Comment:
if the device code or vendor code files don't exist - it's probably the incorrect interface. 
so in this case we just ignore it. 
if none not found - I assume device or vendor is just set to "nil" ? or ""

Reply:
Added extra check for this to consider the case when device or vendor code does not exist